spec.md

```diff
@@ -547,212 +547,129 @@ See [@sec:mem] for more details.
 
 # Types
 
-FIRRTL has four classes of types: _ground_ types, _aggregate_ types, _reference_ types, and _property_ types.
-Ground types are fundamental and are not composed of other types.
-Aggregate types and reference types are composed of one or more aggregate or ground types.
-Reference types may not contain other reference types.
-Property types represent information about the circuit that is not hardware.
+FIRRTL has four classes of types: **ground** types, **aggregate** types, **probe** types, and **property** types.
 
 ## Ground Types
 
-There are five classes of ground types in FIRRTL: unsigned integer types, signed integer types, a clock type, reset types, and analog types.
+Ground types are the most fundamental types.
+They include integer types, clocks, and resets.
 
 ### Integer Types
 
-Both unsigned and signed integer types may optionally be given a known non-negative integer bit width.
+FIRRTL supports both signed and unsigned integer types.
 
-``` firrtl
-UInt ; unsigned int type with inferred width
-SInt ; signed int type with inferred width
-UInt<10> ; unsigned int type with 10 bits
-SInt<32> ; signed int type with 32 bits
-```
-
-Alternatively, if the bit width is omitted, it will be automatically inferred by FIRRTL's width inferencer, as detailed in [@sec:width-inference].
-
-#### Zero Bit Width Integers
+`UInt<n>` is an `n`-bit wide unsigned integer.
+`SInt<n>` is an `n`-bit wide signed integer.
 
-Integers of width zero are permissible.
-They are always zero extended.
-Thus, when used in an operation that extends to a positive bit width, they behave like a zero.
-While zero bit width integers carry no information, we allow 0-bit integer constant zeros for convenience: `UInt<0>(0)`{.firrtl} and `SInt<0>(0)`{.firrtl}.
-
-``` firrtl
-wire zero_u : UInt<0>
-invalidate zero_u
-wire zero_s : SInt<0>
-invalidate zero_s
-
-wire one_u : UInt<1>
-connect one_u, zero_u
-wire one_s : SInt<1>
-connect one_s, zero_s
 ```
-
-Is equivalent to:
-
-``` firrtl
-wire one_u : UInt<1>
-connect one_u, UInt<1>(0)
-wire one_s : SInt<1>
-connect one_s, SInt<1>(0)
+UInt<10> ; a 10-bit unsigned integer
+SInt<32> ; 32-bit signed integer
 ```
 
-### Clock Type
-
-The clock type is used to describe wires and ports meant for carrying clock signals.
-The usage of components with clock types are restricted.
-Clock signals cannot be used in most primitive operations, and clock signals can only be connected to components that have been declared with the clock type.
-
-The clock type is specified as follows:
-
-``` firrtl
-Clock
-```
+Both `UInt<0>`{.firrtl} and `SInt<0>`{.firrtl} are valid types.
+They are considered to have only a single value representing zero (written as either `UInt<0>(0)`{.firrtl} or `SInt<0>(0)`{.firrtl}).
+They are always zero extended, and thus, behave like zero when they are extended to a positive width.
 
-### Reset Type
+TODO: Talk about how zero-bit integers are lowered. And how they can be stripped out.
```

Comment:
The PR should be careful to avoid putting ABI information into the spec. E.g., this discussion could go in the direction of what happens to zero-width ports. However, that is part of the ABI guarantees of the port lowering ABIs.

There is something needed here, though!

Reply:
Agreed. There was a whole paragraph on this, and I stripped it out. 

I'll put it in my notes to add to the ABI doc and remove this TODO.